pandas/tests/dtypes/test_inference.py

```diff
@@ -507,6 +507,13 @@ def test_convert_numeric_int64_uint64(self, case, coerce):
         result = lib.maybe_convert_numeric(case, set(), coerce_numeric=coerce)
         tm.assert_almost_equal(result, expected)
 
+    def test_convert_numeric_string_uint64(self):
+        # GH32394
+        result = lib.maybe_convert_numeric(
+            np.array(["uint64"], dtype=object), set(), coerce_numeric=True
+        )
+        assert np.isnan(result)
```

Comment:
this is not a good test - does it replicate what the OP was? this is not user facing at all

pls follow what is going on in other tests

Reply:
The OP currently takes this path, though it'd be good to test at the `pd.to_numeric` level too.